src/Badge.js

```diff
@@ -25,7 +25,7 @@ class Badge extends React.Component {
 
         return React.cloneElement(element, {
             className: 'mdl-badge',
-            'data-badge': this.props.text
+            'data-badge': this.props.text || undefined
```

Comment:
Sorry, actually we can't, undefined equals string 'undefined' if it's converted to string.